heron/uploaders/src/java/com/twitter/heron/uploader/s3/S3Uploader.java

```diff
@@ -79,52 +84,68 @@ public void initialize(Config config) {
     bucket = S3Context.bucket(config);
     String accessKey = S3Context.accessKey(config);
     String accessSecret = S3Context.secretKey(config);
-    String uri = S3Context.uri(config);
+    String awsProfile = S3Context.awsProfile(config);
+    String proxy = S3Context.proxyUri(config);
+    String endpoint = S3Context.uri(config);
     String customRegion = S3Context.region(config);
+    AmazonS3ClientBuilder builder = AmazonS3ClientBuilder.standard();
 
     if (bucket == null || bucket.isEmpty()) {
       throw new RuntimeException("Missing heron.uploader.s3.bucket config value");
     }
 
-    if (accessKey == null || accessKey.isEmpty()) {
-      throw new RuntimeException("Missing heron.uploader.s3.access_key config value");
-    }
+    // If an accessKey is specified, use it. Otherwise check if an aws profile
+    // is specified. If neither was set just use the DefaultAWSCredentialsProviderChain
+    // by not specifying a CredentialsProvider.
+    if ((accessKey != null && !accessKey.isEmpty())
```

Comment:
If we add a helper method in this class to do `foo != null && !foo.isEmpty()` it would improve the readability of this code:

```
private static boolean isEmpty(String... value) { ... }
```

We could then simplify these conditionals to do:
```
if (!isEmpty(accessKey, accessSecret) {
  // set credentials
} else if (!isEmpty(awsProfile) {
  // set creentials
} else ...
```